### PR TITLE
Clear Branding Resolver Cache Hierarchies when Unpublishing or Deleting Branding Preferences

### PR DESCRIPTION
**Pull request by dhaura**
### Purpose
- Clean resolver cache hierarchies when upublishing or deleting branding/ custom text preferences.
- Refactor branding preference manager unit tests.
    - Remove `MockUIBrandingPreferenceResolver` since the implementation of `UIBrandingPreferenceResolver` is now available in the resolver component.

### Related Issues
- https://github.com/wso2/product-is/issues/20616

### CODE REVIEW

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java [hunk 1]
```diff
@@ -702,4 +710,34 @@ private void validatePreferenceUrls(BrandingPreference preference) throws Brandi
...
+     */
+    private void clearBrandingResolverCacheIfRequired(BrandingPreference previousBrandingPreference,
+                                                      BrandingPreference updatedBrandingPreference, String tenantDomain)
+            throws BrandingPreferenceMgtException {
+
+        JSONObject previousPreference = new JSONObject((LinkedHashMap) previousBrandingPreference.getPreference());
+        JSONObject updatedPreference = new JSONObject((LinkedHashMap) updatedBrandingPreference.getPreference());
+
+        // If configs.isBrandingEnabled is not found in preferences,  it is assumed that branding is enabled by default.
+        boolean isPreviousPreferencesPublished = !previousPreference.has("configs") ||
```
Comment (SujanSanjula96) [inline components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java:730]: shall we use constants for `configs ` and `isBrandingEnabled `

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java [hunk 2]
```diff
@@ -702,4 +710,34 @@ private void validatePreferenceUrls(BrandingPreference preference) throws Brandi
...
+                previousPreference.getJSONObject("configs").optBoolean("isBrandingEnabled", true);
+        boolean isUpdatedPreferencesPublished = !updatedPreference.has("configs") ||
+                updatedPreference.getJSONObject("configs").optBoolean("isBrandingEnabled", true);
+
+        if (isPreviousPreferencesPublished == isUpdatedPreferencesPublished) {
+            return;
+        }
+
+        BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver()
```
Comment (SujanSanjula96) [inline components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java:739]: Thought that  having a private method to return `BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver()` will make code more readable. WDYT?

Comment (dhaura) [reply components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java]: Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/41/commits/72cd2eb7d53698c795918b25058de2421d59dfbd

File: components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java [hunk 3]
```diff
@@ -666,6 +667,11 @@ private void clearCustomTextResolverCacheHierarchy(OrganizationManager organizat
 
     private void clearCustomTextResolverCache(String tenantDomain, String organizationId, String resourceName) {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java:668]: Better to have a separate test resolver cache clear method just passing the tenant domain and org id(if needed) but without a resourceName.

Then the idea is clear that, if someone wants to clear all the records in particular cache respective to that org need to invoke that method.

Comment (dhaura) [reply components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java:696]: Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/41/commits/aa0dfd886f8e0226f967d15dd07e3c4e49544a27

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**